### ISSUE TITLE
[spiflash] Unable to flash examples to Nexys FPGA

### ISSUE THREAD

**Issue by alistair23** [lowRISC/opentitan#3315]
Since https://github.com/lowRISC/opentitan/pull/2664 has been merged I am unable to flash any binaries to the Nexys FPGA.

When flashing a binary to the Nexys FPGA before the PR mentioned above I could run:

```shell
build-out/sw/host/spiflash/spiflash --input=build-out/sw/device/examples/hello_world/hello_world_fpga_nexysvideo.bin
```

After that PR I just get this error instead:

```
Unable to open FTDI SPI interface.
```

**Comment by alistair23** [comment]
PS: Looking at the code it looks like `--dev_id` and `--dev_sn` are now required, although that isn't mentioned in the spiflash help or the error message. It also isn't mentioned in the documentation: https://github.com/moidx/opentitan/blob/b9fea494e568ce7ab023e8818187f23dc88545d7/sw/host/spiflash/README.md#run-the-tool-in-fpga

**Comment by imphil** [comment]
Yes, I've seen the same issue (and thought it was only me). Would you mind doing a PR to fix it (either the docs, or the code).

**Comment by moidx** [comment]
This seems to be an unintended side effect of #2664. I can try to fix this if no one else is looking at implementing a fix. PLMK.

**Comment by alistair23** [comment]
I don't think anyone else is looking into it, so go at it :)

**Comment by kianbeng-lim** [comment]
I had the same problem. I had to initialize the Options struct to get the code to work without the --dev_id and --dev_sn option.

struct Options {
int32_t device_vendor_id = 0 ; //Init this to match the check in the code
int32_t device_product_id = 0;


  if (options_.device_vendor_id == 0 || options_.device_product_id == 0) {
    ctx = MPSSE(/*mode=*/SPI0, options_.spi_frequency, /*endianess=*/MSB);
  }

**Comment by mdhayter** [comment]
I got it to work for FTDI on nexysvideo with #3358 but I didn't test for more than that use case.

**Comment by moidx** [comment]
Fixed in lowrisc/opentitan#3358. Thanks @mdhayter!
